### ISSUE TITLE
tserver crash when copying data

### ISSUE THREAD

**Issue by nickhristov** [YugaByte/yugabyte-db#77]
Encountered a tserver crash today when I tried to use copy command. In this particular case the data was aggregate counters. So, effectively in the same batch we had multiple records which had the same primary key, and non-primary key columns in the table were all counters.

Error from tserver logs:

```
E0305 18:37:00.424288  6387 process_context.cc:180] SQL Error: Execution Error. Reading from a row modified in the same batch request is not supported yet
```

**Comment by kmuthukk** [comment]
Hi Robert:

Additional information on this issue. 

They are using:
```
$ ~/tserver/bin/yb-tserver --version
version 0.9.5.0 build 25 build_type RELEASE
```
Stack trace from core file:
```
Core was generated by `/home/yugabyte/tserver/bin/../lib/unwrapped/yb-tserver --flagfile /home/yugabyt'.
Program terminated with signal 6, Aborted.
#0  0x00007f289e22c198 in raise () from /home/yugabyte/tserver/bin/../lib/linuxbrew/libc.so.6
(gdb) where
#0  0x00007f289e22c198 in raise () from /home/yugabyte/tserver/bin/../lib/linuxbrew/libc.so.6
#1  0x00007f289e22d5ea in abort () from /home/yugabyte/tserver/bin/../lib/linuxbrew/libc.so.6
#2  0x00007f289ed5839e in __gnu_cxx::__verbose_terminate_handler () at ../../../../libstdc++-v3/libsupc++/vterminate.cc:95
#3  0x00007f289ed56c52 in __cxxabiv1::__terminate (handler=<optimized out>) at ../../../../libstdc++-v3/libsupc++/eh_terminate.cc:47
#4  0x00007f289ed56c88 in std::terminate () at ../../../../libstdc++-v3/libsupc++/eh_terminate.cc:57
#5  0x00007f289ed56e8e in __cxxabiv1::__cxa_rethrow () at ../../../../libstdc++-v3/libsupc++/eh_throw.cc:125
#6  0x00007f28a8d91a43 in yb::ql::GramProcessor::parse (this=this@entry=0x654cb5b0) at /opt/builds/yugabyte/build/release-gcc-dynamic-enterprise/src/yb/yql/cql/ql/parser/parser_gram.y.cc:30019
#7  0x00007f28a8d87518 in yb::ql::Parser::Parse (this=this@entry=0x654cb1f0, ql_stmt=..., reparsed=<optimized out>, mem_tracker=...) at ../../../../../../../../src/yb/yql/cql/ql/parser/parser.cc:52
#8  0x00007f28a94dfc6d in yb::ql::QLProcessor::Parse (this=this@entry=0x654cb100, ql_stmt=..., parse_tree=parse_tree@entry=0x7f2825469640, reparsed=reparsed@entry=false, mem_tracker=...)
   at ../../../../../../../src/yb/yql/cql/ql/ql_processor.cc:131
#9  0x00007f28a94dfe4d in yb::ql::QLProcessor::Prepare (this=this@entry=0x654cb100, ql_stmt=..., parse_tree=parse_tree@entry=0x7f2825469640, reparsed=<optimized out>, mem_tracker=...)
   at ../../../../../../../src/yb/yql/cql/ql/ql_processor.cc:159
#10 0x00007f28a94e0187 in yb::ql::QLProcessor::RunBatch (this=this@entry=0x654cb100, ql_stmt=..., params=..., parse_tree=parse_tree@entry=0x7f2825469640, reparsed=<optimized out>)
   at ../../../../../../../src/yb/yql/cql/ql/ql_processor.cc:209
#11 0x00007f28a9fb3f98 in yb::cqlserver::CQLProcessor::ProcessRequest (this=0x654cb100, req=...) at ../../../../../../../src/yb/yql/cql/cqlserver/cql_processor.cc:336
#12 0x00007f28a9fb0265 in yb::cqlserver::CQLProcessor::ProcessRequest (this=this@entry=0x654cb100, req=...) at ../../../../../../../src/yb/yql/cql/cqlserver/cql_processor.cc:209
#13 0x00007f28a9fb069a in yb::cqlserver::CQLProcessor::ProcessCall (this=this@entry=0x654cb100, call=...) at ../../../../../../../src/yb/yql/cql/cqlserver/cql_processor.cc:161
#14 0x00007f28a9fcbe58 in yb::cqlserver::CQLServiceImpl::Handle (this=0x2961ba0, inbound_call=...) at ../../../../../../../src/yb/yql/cql/cqlserver/cql_service.cc:130
#15 0x00007f28a3a6ce27 in Handle (incoming=..., this=0x31435f0) at ../../../../../src/yb/rpc/service_pool.cc:191
#16 Run (this=<optimized out>) at ../../../../../src/yb/rpc/service_pool.cc:206
#17 yb::rpc::TasksPool<yb::rpc::(anonymous namespace)::InboundCallTask>::WrappedTask::Run (this=<optimized out>) at ../../../../../src/yb/rpc/tasks_pool.h:70
#18 0x00007f28a3a6f279 in yb::rpc::(anonymous namespace)::Worker::Execute (this=0x3a5f2d0) at ../../../../../src/yb/rpc/thread_pool.cc:98
#19 0x00007f28a20fd6f6 in operator() (this=0x19af3958) at /home/centos/.linuxbrew-yb-build/Cellar/gcc/5.3.0/include/c++/5.3.0/functional:2271
#20 yb::Thread::SuperviseThread (arg=<optimized out>) at ../../../../../src/yb/util/thread.cc:602
#21 0x00007f289e59f314 in start_thread () from /home/yugabyte/tserver/bin/../lib/linuxbrew/libpthread.so.0
#22 0x00007f289e2e0bed in clone () from /home/yugabyte/tserver/bin/../lib/linuxbrew/libc.so.6
```

**Comment by robertpang** [comment]
@nickhristov We have just removed the implementation restriction "Reading from a row modified in the same batch request is not supported yet". The fix is in commit [cf5c2f2](https://github.com/YugaByte/yugabyte-db/commit/cf5c2f2784e795ce26b4f4fbbe3c729854287532). May I ask you to upgrade to a newer build that has the commit and try again? Thanks.
